wsproto/handshake.py

```diff
@@ -429,22 +438,24 @@ def server_extensions_handshake(
         for extension in supported:
             if extension.name == name:
                 accept = extension.accept(offer)
-                if accept is True:
-                    accepts[extension.name] = True
-                elif accept is not False and accept is not None:
-                    accepts[extension.name] = accept.encode("ascii")  # type: ignore
+                if isinstance(accept, bool):
+                    if accept:
+                        accepts[extension.name] = True
+                elif accept is not None:
+                    accepts[extension.name] = accept.encode("ascii")
 
     if accepts:
         extensions: List[bytes] = []
         for name, params in accepts.items():
-            name = name.encode("ascii")  # type: ignore
-            if params is True:
-                extensions.append(name)  # type: ignore
+            name_bytes = name.encode("ascii")
+            if isinstance(params, bool):
+                assert params
```

Comment:
I'm not sure these two lines equate to `if params is True` - if the assert is optimised away wouldn't `params = False` also satisfy this?

Reply:
The assert here should never trigger, because the loop above (the one that generates `accepts`) never assigns `False`, only `True`.

The reason I changed this code is that mypy really needs `isinstance` to disambiguate unions. It doesn't understand that if a value with type `Union[bool, bytes]` satisfies both `is not True` and `is not False`, then it is not a bool.